### PR TITLE
added matrix inverse + test

### PR DESCRIPTION
**Pull request by erdieee**
## Summary
Added missing method for computing the inverse of a matrix

## Quick changelog
- Matrix inverse can be invoked with bla::Matrix::I()
- Tests for correct computation of the inverse

### CODE REVIEW

File: src/matrix.h [hunk 1]
```diff
@@ -95,6 +96,84 @@ namespace bla
...
+                inv.RowMultiplyByScalar(j, 1 / inv(j, j));
+                for (size_t i = 0; i < NumRows(); i++)
+                {
+                    if (i == j)
+                        continue;
+                    inv.RowMultiplySubtract(i, j, inv(i, j));
+                }
+            }
+
+            Matrix<T, ORD> inverse(dim, dim);
```
Comment (shirnschall) [inline src/matrix.h:166]: If we declare the matrix inv with dim (n,2m) we should do column operations. This way the data vector will contain first all elements of the original matrix and then all identity matrix elements. After gauss jordan, inv would contain first the identity and then all inverse elements. This way we do not have to copy the matrix again in memory but we can split the data array in the middle. e.g. replace \*data with \*data+m\*n.

Likewise, for row operations it seems to be better to initialize inv with dim (2n,m). 

This should also be true after slicing is implemented.

**Review by shirnschall — CHANGES_REQUESTED** [review]

**Comment by shirnschall** [comment]
please fix the typos before merging

File: src/matrix.h [hunk 2]
```diff
@@ -95,6 +96,84 @@ namespace bla
             }
         }
 
+        void RowMultiplyByScalar(size_t row, T s)
+        {
+            for (size_t i = 0; i < NumCols(); i++)
+                (*this)(row, i) *= s;
+        }
+
+        void RowAddRow(size_t row, T to)
```
[the remark this answers is not in the document]
Comment (erdieee) [reply src/matrix.h]: nice catch 👍, fixed

File: src/matrix.h [hunk 3]
```diff
@@ -95,6 +96,84 @@ namespace bla
...
+                if ((*this)(i, row) != 0)
+                    break;
+            }
+            if (i == NumRows())
+                throw std::invalid_argument("Matrix is singular");
+            if (i != row)
+                SwapRows(i, row);
+        }
+
+        void SwapRows(size_t a, size_t b)
```
[the remark this answers is not in the document]
Comment (erdieee) [reply src/matrix.h]: valid point. implemented the second option.

File: src/matrix.h (hunk 1 above)
Comment (shirnschall) [reply src/matrix.h:168]: O.K. for now 👍

**Review by shirnschall — APPROVED** [review]

**Comment by shirnschall** [comment]
LGTM

**Pull request merged**